daemon/cmd/state.go

```diff
@@ -304,7 +305,39 @@ func (d *Daemon) regenerateRestoredEndpoints(state *endpointRestoreState) (resto
 		}
 	}
 
+	if option.Config.EnableIPSec &&
+		(option.Config.IPAM == ipamOption.IPAMENI || option.Config.IPAM == ipamOption.IPAMAzure) {
+		// If IPsec is enabled on EKS or AKS, we need to restore the host
+		// endpoint before any other endpoint, to ensure a dropless upgrade.
+		// This code can be removed in v1.15.
```

Comment:
Is there a specific PR(s) that allows this? Probably worth specifying those.

Reply:
The code removal isn't being enabled by any specific PR. We can remove in v1.15 because by then we will be sure that nobody needs to up/downgrade from/to a version of Cilium with the old XFRM configs. The old XFRM configs only exist in v1.13 and prior versions.